app/controllers/api/journeys_controller.rb

```diff
@@ -2,7 +2,9 @@
 
 module Api
   class JourneysController < ApiController
+    before_action :validate_required_idempotency_key
     before_action :validate_params, only: %i[create update]
+    around_action :idempotent_action
```

Comment:
I think we might need to only do this on create and update, as the other requests being returned cached might be a  problem and there isn't a risk in get/show actions

Reply:
interesting point, I had been steering the suppliers to provide an idempotency key for _all_ requests; however in the latest commit I no longer check the key for show or index on the journey events controller

Reply:
The code has been updated to only do the idempotency check for `:create` and `:update` in the `journeys_controller`